src/d/actor/d_a_npc_cb1.cpp

```diff
@@ -72,532 +88,2993 @@ static dCcD_SrcCyl l_wind_cyl_src = {
+    /* Nonmatching - instruction ordering */
+    if (calcTiming == J3DNodeCBCalcTiming_In) {
+        J3DJoint* joint = (J3DJoint*)node;
+        J3DModel* model = j3dSys.getModel();
+
+        daNpc_Cb1_c* a_this = (daNpc_Cb1_c*)model->getUserArea();
+        if (a_this != NULL) {
+            static cXyz l_nutOffset(0.0f, 0.0f, 0.0f);
+            static cXyz l_nutBase(1.0f, 0.0f, 0.0f);
+
+            s32 jnt_no = joint->getJntNo();
+
+            cXyz temp2(model->getAnmMtx(jnt_no)[0][3], model->getAnmMtx(jnt_no)[1][3], model->getAnmMtx(jnt_no)[2][3]);
+            a_this->getNutPos() += a_this->getNusSpeed();
+            cXyz temp = a_this->getNutPos() - temp2;
+
+            f32 temp4 = temp.abs();
+            if(!cM3d_IsZero(temp4)) {
+                temp *= 14.0f / temp4;
+            }
+
+            a_this->getNutPos() = temp2 + temp;
+
+            a_this->getNusSpeed() += temp * -0.15f;
+            a_this->getNusSpeed().y += -1.3f;
+
+            f32 temp5 = a_this->getNusSpeed().abs();
+            if(!cM3d_IsZero(temp5)) {
+                a_this->getNusSpeed() *= 5.8f / temp5;
+            }
+
+            Mtx temp3;
+            cM3d_UpMtx_Base(l_nutBase, temp, temp3);
+            model->getAnmMtx(jnt_no)[0][0] = temp3[0][0];
+            model->getAnmMtx(jnt_no)[0][1] = temp3[0][1];
+            model->getAnmMtx(jnt_no)[0][2] = temp3[0][2];
+            model->getAnmMtx(jnt_no)[1][0] = temp3[1][0];
+            model->getAnmMtx(jnt_no)[1][1] = temp3[1][1];
+            model->getAnmMtx(jnt_no)[1][2] = temp3[1][2];
+            model->getAnmMtx(jnt_no)[2][0] = temp3[2][0];
+            model->getAnmMtx(jnt_no)[2][1] = temp3[2][1];
+            model->getAnmMtx(jnt_no)[2][2] = temp3[2][2];
+            cMtx_copy(model->getAnmMtx(jnt_no), J3DSys::mCurrentMtx);
+        }
+    }
+
+    return TRUE;
 }
 
+static fpc_ProcID l_msgId;
+static msg_class* l_msg;
+
 /* 00000CF4-00000D9C       .text ppNodeCallBack__FP7J3DNodei */
-static BOOL ppNodeCallBack(J3DNode*, int) {
-    /* Nonmatching */
+static BOOL ppNodeCallBack(J3DNode* node, int calcTiming) {
+    if (calcTiming == J3DNodeCBCalcTiming_In) {
+        J3DJoint* joint = (J3DJoint*)node;
+        J3DModel* model = j3dSys.getModel();
+
+        daNpc_Cb1_c* a_this = (daNpc_Cb1_c*)model->getUserArea();
+        if (a_this != NULL) {
+            s32 jnt_no = joint->getJntNo();
+            mDoMtx_stack_c::YrotS(a_this->getWork3());
+            mDoMtx_stack_c::revConcat(model->getAnmMtx(jnt_no));
+            cMtx_copy(mDoMtx_stack_c::get(), J3DSys::mCurrentMtx);
+            model->setAnmMtx(jnt_no, mDoMtx_stack_c::get());
+        }
+    }
+
+    return TRUE;
 }
 
 /* 00000D9C-00001458       .text createHeap__11daNpc_Cb1_cFv */
-void daNpc_Cb1_c::createHeap() {
-    /* Nonmatching */
+BOOL daNpc_Cb1_c::createHeap() {
+    J3DModelData* modelData = (J3DModelData*)dComIfG_getObjectRes("Cb", CB_BDL_CB);
+    JUT_ASSERT(0x3A8, modelData != 0);
```

Comment:
```suggestion
    JUT_ASSERT(0x3A8, modelData != NULL);
```
These asserts can use defines in them and they get converted to their value in the string itself. Also applies to the other `modelData != 0` asserts in this TU.